### PR TITLE
Migrate from ansi_term to anstyle

### PR DESCRIPTION
**Pull request by nbdd0121**
ansi_term is unmaintained. In a range of possible alternatives, anstyle was chosen because it's a dependency of clap already.

### CODE REVIEW

File: src/json.rs [hunk 1]
```diff
@@ -459,7 +459,9 @@ impl JsonEmitter {
                 write!(
                     w,
                     "{}",
-                    self.color.comment.paint(format!("{} {}", leader, line))
+                    self.color
+                        .comment
+                        .paint(format_args!("{} {}", leader, line))
```
Comment (jwnrt) [inline src/json.rs:464]: is `format_args` quicker when the result is going to be an argument of another formatter?

**Review by jwnrt — APPROVED** [review]
LGTM

File: src/json.rs (hunk 1 above)
Comment (nbdd0121) [reply src/json.rs:464]: Yes, it allows formatting into final destination directly without going through another `String`.

Comment (HU90m) [reply src/json.rs:464]: It's super cool. I was unaware of this.

**Review by HU90m — APPROVED** [review]

**Pull request merged**